### DISCUSSION TITLE
Error: The deploy step encountered an error: The process '/usr/bin/git' failed with exit code 128 ❌ Notice: Deployment failed! ❌

### DISCUSSION THREAD

**Discussion by RunningLiLi** [JamesIves/github-pages-deploy-action#1421] [Q&A] [answered]
It works fine in the forked repository, but once pr to the repository it gives this error
![image](https://github.com/JamesIves/github-pages-deploy-action/assets/93420481/16861f7a-565d-4a5c-8bdd-41e04219b943)
I try to do this
![image](https://github.com/JamesIves/github-pages-deploy-action/assets/93420481/aed10bff-5deb-431c-818b-52f716d6dc2d)
and
![image](https://github.com/JamesIves/github-pages-deploy-action/assets/93420481/f8e37eab-f3b8-414f-8fc3-b846b0977ee1)
But it's still not working.

**Comment by RunningLiLi** [comment]
I can't understand why there are two structures for the same configuration

**Comment by JamesIves** [comment]
Can you provide the full workflow file and the error log? The error you're pasting isn't actually the _real_ error, there should be something a couple lines above that will reveal the point of failure.

**Comment by buzcarter** [comment]
Am getting the same error (attached workflow file)

```
Switched to a new branch 'github-pages-deploy-action/byvtcv3rf'
/usr/bin/git commit -m Deploying to gh-pages from @ buzcarter/song-formatter@09b0b18083e9958fe6c7703eb4fb6d6f1261015e 🚀 --quiet --no-verify
Force-pushing changes...
/usr/bin/git push --force ***github.com/buzcarter/song-formatter.git github-pages-deploy-action/byvtcv3rf:gh-pages
remote: Permission to buzcarter/song-formatter.git denied to github-actions[bot].
fatal: unable to access 'https://github.com/buzcarter/song-formatter.git/': The requested URL returned error: 403
Running post deployment cleanup jobs… 🗑️
/usr/bin/git checkout -B github-pages-deploy-action/byvtcv3rf
Reset branch 'github-pages-deploy-action/byvtcv3rf'
/usr/bin/chmod -R +rw github-pages-deploy-action-temp-deployment-folder
/usr/bin/git worktree remove github-pages-deploy-action-temp-deployment-folder --force
Error: The deploy step encountered an error: The process '/usr/bin/git' failed with exit code 128 ❌
Notice: Deployment failed! ❌

Warning: The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
```
[publishDemo.yml.txt](https://github.com/JamesIves/github-pages-deploy-action/files/12566494/publishDemo.yml.txt)

**Reply by JamesIves** [reply, marked as answer]
Please provide your workflow file and validate the permissions outlined in the readme.

**Reply by buzcarter** [reply]
My workflow is attached above, but it's fixed: it was permissions (ahem, user error/RTFM)

Thank you
